src/scripts/ping-okta-server.ts

```diff
@@ -0,0 +1,103 @@
+import * as TE from 'fp-ts/lib/TaskEither';
+import * as E from 'fp-ts/lib/Either';
+import { RootCommand } from '..';
+import * as Console from 'fp-ts/lib/Console';
+import { pipe } from 'fp-ts/lib/function';
+import { Argv } from 'yargs';
+import { oktaReadOnlyClient } from './services/client-service';
+import { pingOktaServer, validateCredentials } from './services/okta-service';
+
+/**
+ * Pings the okta server to see if it is up and running.
+ * @param clientId - the client id of the okta application.
+ * @param organisationUrl - the url of the okta organisation.
+ * @returns a TaskEither that resolves to a string if the okta server is up and running, otherwise an error message.
+ */
+export const validateOktaServerIsRunning = (
+  clientId: string,
+  organisationUrl: string
+): TE.TaskEither<Error, 'Okta server is up and running.'> =>
+  pipe(
+    Console.info('Pinging okta server...'),
+    TE.rightIO,
+    // eslint-disable-next-line functional/functional-parameters
+    TE.chain(() =>
+      TE.chain(
+        ({
+          organisationUrl,
+          clientId,
+        }: {
+          readonly organisationUrl: string;
+          readonly clientId: string;
+        }) => pingOktaServer(clientId, organisationUrl)
+      )(TE.right({ organisationUrl, clientId }))
+    ),
+    // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+    TE.chain((response) => {
```

Comment:
**Issue:** (sorry this feedback is late) move this inside `pingOktaServer`. The code that interprets the details of the response from Okta should be inside the service.

Reply:
modified in commit: d29e26e06d56cf9078a5ad96037d056001000970